File: Excel/900-999/926/926 Challenge.R
library(tidyverse)
library(readxl)

path <- "900-999/926/926 Conversion to ddhhmmss Format.xlsx"
input <- read_excel(path, range = "A1:A23")
test <- read_excel(path, range = "B1:B23")

parse_time <- function(x) {
  str_match(x, "(?:(\\d+)d)?(?:(\\d+)h)?(?:(\\d+)m)?(?:(\\d+)s)?") |>
    as_tibble(.name_repair = "minimal") |>
    set_names(c("all", "d", "h", "m", "s")) |>
    select(-all) |>
    mutate(across(everything(), ~ replace_na(as.numeric(.x), 0))) |>
    transmute(
      total = d * 86400 + h * 3600 + m * 60 + s
    ) |>
    mutate(
      sprintf(
        "%02d.%02d:%02d:%02d",
        total %/% 86400,
        (total %% 86400) %/% 3600,
        (total %% 3600) %/% 60,
        total %% 60
      )
    ) |>
    pull()
}

result = input %>%
  mutate(parsed = map_chr(Data, parse_time))

all.equal(result$parsed, test$`Answer Expected`)
# [1] TRUE

File: Excel/900-999/926/926 Challenge.py
import pandas as pd
import re

path = "900-999/926/926 Conversion to ddhhmmss Format.xlsx"
input = pd.read_excel(path, usecols="A", nrows=23)
test = pd.read_excel(path, usecols="B", nrows=23)

def parse_time(x):
    t = pd.to_timedelta(re.sub(r'(\d+)([dhms])',
                               lambda m: f"{m.group(1)} {'days hours minutes seconds'.split()[ 'dhms'.index(m.group(2)) ]}",
                               x))
    d = t.days
    h = t.seconds // 3600
    m = (t.seconds % 3600) // 60
    s = t.seconds % 60
    return f"{d:02}.{h:02}:{m:02}:{s:02}"

result = input['Data'].apply(parse_time)

print(result.equals(test['Answer Expected']))